### ISSUE TITLE
ClassCastException with native event when using autocomplete

### ISSUE THREAD

**Issue by thubalek** [JetBrains/compose-jb#765]
Hi, I'm working with JetPack Compose for Web and I found that following snippet

```kotlin
@Composable
fun ClassCastExceptionTestCaseView() {
    Input(attrs = {
        onInput {
            console.log("New value is", it.nativeEvent.target.asDynamic().value as String)
        }
        attr("type", "email")
        attr("autocomplete", "email")
    }, value = "")
}
```

causes `ClassCastException` when using [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete). When typing then there is no problem.

In debugger it looks like this:

![image](https://user-images.githubusercontent.com/1260262/121330699-8acbaf00-c916-11eb-80ef-5e91a56c6af1.png)

I'm not sure but it may be a bug in JetPack Compose for Web.

**Comment by thubalek** [comment]
According suggestion on Kotlin Slack I implemented this workaround:

```kotlin
@Composable
fun ClassCastExceptionTestCaseView() {
    Input(attrs = {
        // onInput {
        //     console.log("New value is", it.nativeEvent.target.asDynamic().value as String)
        // }
        attr("type", "email")
        attr("autocomplete", "email")
    }, value = "") {
        DomSideEffect {
          val callback : ((Event)->Unit) = { event ->
                val newText = event.target.asDynamic().value
                console.log("New value is", newText)
            }
            it.addEventListener("input", callback)
            onDispose { element ->
                element.removeEventListener("input", callback)
            }
         }
    }
} 
```

**Comment by hfhbd** [comment]
This works too
```kotlin
Input(...
  attrs = {
    addEventListener("input") {
      val target = it.nativeEvent.target as HTMLInputElement
      onChange(target)
    }
  }
)
```

**Comment by eymar** [comment]
fixed in 0.5.0-build228

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6402">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
